nautobot/utilities/views.py

```diff
@@ -132,3 +166,1046 @@ def get_return_url(self, request, obj=None):
+            ),
+        )
+
+    def get_object_for_delete(self, kwargs):
+        # Look up object by slug if one has been provided. Otherwise, use PK.
+        if "slug" in kwargs:
+            return get_object_or_404(self.object_delete_queryset, slug=kwargs["slug"])
+        else:
+            return get_object_or_404(self.object_delete_queryset, pk=kwargs["pk"])
+
+    def handle_object_delete_get(self, request, **kwargs):
+        obj = self.get_object_for_delete(kwargs)
+        form = ConfirmationForm(initial=request.GET)
+
+        return render(
+            request,
+            self.object_delete_template_name,
+            {
+                "obj": obj,
+                "form": form,
+                "obj_type": self.object_delete_queryset.model._meta.verbose_name,
+                "return_url": self.get_return_url(request, obj),
+            },
+        )
+
+    def handle_object_delete_post(self, request, **kwargs):
+        logger = logging.getLogger("nautobot.views.ObjectDeleteView")
+        obj = self.get_object_for_delete(kwargs)
+        form = ConfirmationForm(request.POST)
+
+        if form.is_valid():
+            logger.debug("Form validation was successful")
+
+            try:
+                obj.delete()
+            except ProtectedError as e:
+                logger.info("Caught ProtectedError while attempting to delete object")
+                handle_protectederror([obj], request, e)
+                return redirect(obj.get_absolute_url())
+
+            msg = "Deleted {} {}".format(self.object_delete_queryset.model._meta.verbose_name, obj)
+            logger.info(msg)
+            messages.success(request, msg)
+
+            return_url = form.cleaned_data.get("return_url")
+            if return_url is not None and is_safe_url(url=return_url, allowed_hosts=request.get_host()):
+                return redirect(return_url)
+            else:
+                return redirect(self.get_return_url(request, obj))
+
+        else:
+            logger.debug("Form validation failed")
+
+        return render(
+            request,
+            self.object_delete_template_name,
+            {
+                "obj": obj,
+                "form": form,
+                "obj_type": self.object_delete_queryset.model._meta.verbose_name,
+                "return_url": self.get_return_url(request, obj),
+            },
+        )
+
+
+class BulkImportViewMixin(GetReturnURLMixin, NautobotRouterMixin):
+    """
+    Import objects in bulk (CSV format).
+    queryset: Base queryset for the model
+    model_form: The form used to create each imported object
+    table: The django-tables2 Table used to render the list of imported objects
+    template_name: The name of the template
+    widget_attrs: A dict of attributes to apply to the import widget (e.g. to require a session key)
+    """
+
+    bulk_import_queryset = None
+    bulk_import_model_form = None
+    bulk_import_table = None
```

Comment:
9 times out of 10 the list, bulk-import, bulk-delete, and bulk-edit views will use the same table; the list, bulk-edit, and bulk-delete views will use the same filterset, etc. Again, let's think about how we can minimize the amount of repeated boilerplate.